server/src/main/java/org/elasticsearch/search/aggregations/timeseries/TimeSeriesAggregator.java

```diff
@@ -98,6 +102,14 @@ protected LeafBucketCollector getLeafCollector(LeafReaderContext context, LeafBu
         throws IOException {
         return new LeafBucketCollectorBase(sub, null) {
 
+            @Override
+            public DocIdSetIterator competitiveIterator() throws IOException {
+                if (needCounts) {
+                    return null;
+                }
+                return sub.competitiveIterator();
```

Comment:
I suspect that there is an assumption here that the competitive iterator of the sub aggregation may only be a competitive iterator that visits every TSID. But what if not? Should we OR the sub iterator with an iterator that is guaranteed to visit every TSID? Maybe we can later work on detecting whether we're OR-ing twice the same iterator and dedup them.

Reply:
I extracted two iterators:

- DocsPerOrdIterator: Iterator that visits each ordinal a provided number of times
- DisjunctionDocIdSetIterator: iterator that is the disjunction of the provided iterators.

The timeseries competitive iterator is the disjunction between the subiterators and an iterator that visit each tsid at least once.